src/main/java/com/netflix/imflibrary/st2067_2/ApplicationCompositionFactory.java

```diff
@@ -117,6 +120,9 @@ public static ApplicationComposition getApplicationComposition(ResourceByteRange
                     Constructor<?> constructor = clazz.getConstructor(IMFCompositionPlaylistType.class, Set.class);
                     composition = (ApplicationComposition) constructor.newInstance(imfCompositionPlaylistType, homogeneitySelectionSet);
                     imfErrorLogger.addAllErrors(composition.getErrors());
+                    // ST 2067-203 MGASADMVirtualTrackParameterSet checks
```

Comment:
Could there be a better place to invoke this `IMFMGASADMConstraintsChecker.checkMGASADMVirtualTrackParameterSet()` method? Maybe in the inheritors of the ApplicationComposition class? 

Since this is a factory class that doesn't invoke any other constraints checks, there may be a better, more targeted place to perform these validations.

Reply:
Moved to [src/main/java/com/netflix/imflibrary/app/IMPAnalyzer.java](https://github.com/Netflix/photon/pull/376/files#diff-05d5b910201cd0155c75aa27b81ea7586b373cca61c6db03ef53a55b56ced8b6)